### ISSUE TITLE
Platform-aware titlebar design

### ISSUE THREAD

**Issue by 19lmyers** [sindrets/atom-title-bar-replacer#4]
The current titlebar added by this package doesn't really match the Windows titlebar. Would it be possible to update the design to better fit? (I saw someone has already done something similar [here](https://github.com/Menci/electron-titlebar))

**Comment by sindrets** [comment]
Hello again, @19lmyers! I'm sorry, but I don't quite see where you're coming from with this request. The whole point of this package, as stated in the README, is to move *away* from the Windows title bar, as I feel its colours and design often clashes with most Atom UI themes. Besides, if what you are looking for is the Windows title bar; why don't you just use Atom without this package, as the editor ships with the native Windows title bar on Windows machines by default?

Am I misunderstanding you?

**Comment by 19lmyers** [comment]
The default Windows title bar is provided by the system, and cannot adapt to the theme of the program (like you said). A program-provided title bar (like this one) can, but this one clashes with the design of the other title bars on my system. 

Many applications choose to provide their own title bar (so it can be styled to fit the application) but base the size and button layout on the system specifications.

Simply put, I was hoping for the buttons to behave more like their normal Windows counterparts.

**Comment by 19lmyers** [comment]
[These](https://gist.github.com/19lmyers/2413a61f221a4856e45b31cc004dec2e) custom styles sort of demonstrate what I mean, but aren't exactly what I'm referring to.

**Comment by branden-akana** [comment]
I believe what's being asked here is that he wants the titlebar's design tweaked to closer match Window's toolbar for the sake of consistency with other Windows applications.

While I'm personally not concerned that much of the current design standing out or anything, I think it might be worth considering having the design at least being worked more towards what electron-toolbar has (from the OP):

![](https://github.com/Menci/electron-titlebar/raw/master/windows-screen.png)

Maybe the icons for example?

**Comment by sindrets** [comment]
@19lmyers, @branden-akana I see what you mean. I will seriously consider it for a future update. However, it does raise some difficulties. For instance, the navigation buttons have different sizes based on Windows' UI scale settings, which cannot be accessed from Electron as far as I know. @19lmyers, take for example the custom css you provided: (assuming the buttons had the right size on *your* monitor) they looked ridiculously huge on my 1080p monitor (with 100% on UI scaling). So if I decide to commit to make my package deliver OS consistency, it wouldn't be a very solid implementation if the navigation buttons appeared with the wrong size on a wast amount of monitors and configurations. 

I did play around with the idea of a sort of theming system for the title bar that would let you create multiple profiles (perhaps added to a dropdown menu in the settings), where the user could edit an html template with their own elements and css (including changing the icons). Further there could be a quick access size adjustment in the settings for the nav. buttons that would override the css.

Is this a solution that could work?

@branden-akana: if you want to edit simple things like sizes, paddings, margins and fonts, I already explained a simple way to do this in [another issue](https://github.com/sindrets/atom-title-bar-replacer/issues/3#issuecomment-320788853).

**Comment by branden-akana** [comment]
Sounds good. If anything, I guess a UI scale option could be used to workaround Electron not being able to see it.

By the way, is the `.custom-menu` class made specifically for this package? I wasn't even aware it existed.

**Comment by sindrets** [comment]
@19lmyers, @branden-akana took me some time to get around to this, but the just now published v1.7.0 brings window controls presets, and amongst them is a Windows 10 preset that is almost a pixel perfect recreation of the native Windows 10 controls on 100% UI scaling :)
